### PR TITLE
Fixes #21916 Handle NULL byte injection

### LINKED ISSUE #21916: Handle NULL byte injection in the repository path of a PartItem object.
[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* GlassFish Version (and build number):5.0
* JDK version:1.8
* OS:Linux x86
* Database:

----------

### PR DESCRIPTION
**Pull request by mukumawa**
Fixes #21916

### CODE REVIEW

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — DISMISSED** [review]

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Review by edburns — DISMISSED** [review]
PartItem.java

Line 739 and 753.  Do we trust that getPath() will never return null?

**Review by shingwaichan — CHANGES_REQUESTED** [review]
Please use logging message rather hardcode strange in exception message. See the usage in other file, such as ApplicationContext.java

**Review by edburns — DISMISSED** [review]
Caveat: I am not familiar with this code.  However, I am able to offer general feedback on these changes.  I suggest you check for a null return from getPath() before dereferencing the returned pointer from that method.

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

File: appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java [hunk 1]
```diff
@@ -733,10 +734,30 @@ private void readObject(ObjectInputStream in)
         // read values
         in.defaultReadObject();
 
+        if (repository != null) {
+            if (repository.isDirectory()) {
+                                // Check path for nulls
+                if (repository.getPath() == null || repository.getPath().contains("\0")) {
```
Comment (yaminikb) [inline appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java]: This doesn't seem right. The isDirectory check has succeeded at this point, so getPath() can't be null. You can quickly check by running FindBugs.

Comment (mukumawa) [inline appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java]: I have checked the findbugs locally. I am got findbug here .

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

File: appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java (hunk 1 above)
Comment (shingwaichan) [inline appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java]: If the repository.getPath() is really null, then it should not use the REPOSITORY_PATH_CONTAIN_NULL_CHARACTER message.

**Review by shingwaichan — COMMENTED** [review]
The null check (file != null) for the private method deleteFile seems redundant here. Please double check the usage in PartItem.java.

**Review by shingwaichan — APPROVED** [review]

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by yaminikb** [comment]
Merged after confirming that all tests passed except JDBC which is unrelated to this change.

**Pull request merged**